### DISCUSSION TITLE
Which desktop/web apps work well along side OsmAnd?

### DISCUSSION THREAD

**Discussion by localjo** [osmandapp/OsmAnd#11137] [Q&A]
I'm curious what desktop/web apps people use along side OsmAnd?

As an avid traveler, I'm usually using lists on Google Maps to mark places of interest. They conveniently sync between my computer and my phone, so tracking places I want to go and planning trips is easy to do whether I'm on my phone or my computer. Since OsmAnd doesn't have a desktop app, I'm looking for ways that I can plan trips and track places of interest or favorites from my computer.

I've found a couple of apps like [RouteConverter](https://www.routeconverter.com/) and [BRouter](http://brouter.de/) that I can use for planning routes on my computer and then send those routes to my phone. That's nice for long road trips when I want to plan my route carefully.

I also use https://www.openstreetmap.org/ on my desktop for browsing and editing maps and sharing traces. 

Are there any good desktop apps for syncing bookmarks/favorites/POIs? I would love to find a way to sync my favorites with [NextCloud Maps](https://github.com/nextcloud/maps). There's [an interesting discussion](https://github.com/nextcloud/maps/issues/30) about syncing NC Maps with gnome-maps. I wonder if there would be a relatively easy way to link up OsmAnd's favorites.gpx with NC Maps.

Are there any other desktop apps that pair nicely with OsmAnd for other purposes?

**Comment by vshcherb** [comment]
It's a good question: OsmAnd Android supports Online Routing / Online Maps / B-Router / GPX-file based Routing. Basically you can create your own tiles / vector maps (OsmAndMapCreator - https://docs.osmand.net/en/main@latest/development/map-creation/create-offline-maps-yourself) and even customize routing, it's actually depends a lot what do you mean by Desktop / Web tools and how you want them to be configured with OsmAnd.

**Reply by localjo** [reply]
Thanks for the tip. I think OsmAndMapCreator is a bit more technical than what I am looking for at the moment. I'm looking for apps that someone can use to plan trips and mark favorites, as a traveler, not as a developer.

**Reply by thompehw** [reply]
Have you come across [Offroad](https://sourceforge.net/p/offroadosm/wiki/Home/)? It's a port of OsmAnd to pure Java for desktops. It's in beta, but some people have reported that it's already useful.

Also, it should be possible to run OsmAnd in an Android emulator (e.g. BlueStacks) if your computer can handle it.

**Reply by localjo** [reply]
> Have you come across Offroad

Looks very interesting. I'll give it a try.

>  it should be possible to run OsmAnd in an Android emulator

That is useful for development, but it wouldn't enable many new workflows for the use-case of a traveler who is trying to plan trips and manage favorites, since it's the same OsmAnd UI in a virtual device instead of a physical device. I'm hoping to find some apps that have a native desktop/web UI.

**Reply by realgooseman** [reply]
> Have you come across [Offroad](https://sourceforge.net/p/offroadosm/wiki/Home/)? It's a port of OsmAnd to pure Java for desktops. It's in beta, but some people have reported that it's already useful.
> 
> Also, it should be possible to run OsmAnd in an Android emulator (e.g. BlueStacks) if your computer can handle it.

Sorry for reviving this old thread @localjo @vshcherb @thompehw,

Windows Subsystem for Android has been sunset and will stop working on March 25th, 2025.
All other non-native Android Emulators (like BlueStacks) are bloated and give bad map navigation performance.
Native Windows Maps app got rid of offline maps downloading and usage.
Offroad hasn't been updated in ages, most likely abandoned.

Aren't there any simple offline map viewers for Windows you can think of, like nothing too complicated like QGIS or Grass GIS.
Simple Windows software for offline map viewing (with navigation) like OSMAnd or Organic Maps on android ?

**Comment by clementmas** [comment]
I added support for the OsmAnd Online Tracking feature (Android) on the blogging platform I created: https://travelmap.net/live-tracking
It allows users to automatically update their maps and trace their travel itinerary (this is a paid feature).

I wish the Online Tracking option was available on iOS as well. I'm also getting some issues with users forgetting to change the tracking frequency and sending me thousands of points every few minutes.

But overall this is working great. I also added support for [Wayward](http://wayward.travel) (requires creating an account) and [GPSLogger](https://github.com/mendhak/gpslogger) (requires manual installation).

Do you know other apps capable of sharing the location in real time?

**Comment by johnmcd88** [comment]
I realize this is an older topic, but I've recently been trying to address the same issue - what desktop mapping app to use to complement OSMAnd on my mobile devices. I've looked at Offroad, Viking GPS, JOSM and several others, but they are all either too complex compared to OSMAnd or they don't support a lot of what OSMAnd does that I use a lot (e.g. A/V Notes, customized favorites, etc.). I also want to be able to work offline, since I tend to take my MS Surface tablet on longer trips where I may not have Internet access. What I ended up doing is running OSMAnd on my desktop using BlueStacks (or WSA on Windows 11, although that's still in beta) and using the shared desktop folder to move files between Windows and OSMAnd. After a day of hiking/biking/kayaking/etc. and collecting data points/pics/videos/etc. on my phone I export everything to an OSF file, copy it to the shared folder on my desktop, then import it to OSMAnd running in BlueStacks. I do have to delete everything in OSMAnd on my phone when I start each morning so I don't get duplicates on the desktop, but it works for me.

I can also create a route plan at night on my desktop and copy that to my phone for import.

I thought this might be useful for others grappling with the same issue. Any feedback or suggestions are welcome.

John

**Reply by johnmcd88** [reply]
I realize that this is an older topic, but I've developed an approach for offline maps that somewhat complements OSMAnd. I now use Obsidian, which is a fantastic note taker/organizer app, with the Map View plugin (https://github.com/esm7/obsidian-map-view). The Map View plugin allows you specify your own tile server, so I set up MapsForgeSrv (https://sourceforge.net/projects/mobac/files/MapsforgeSrv/) running on my local system, downloaded the maps I want from MapsForge (https://ftp-stud.hs-esslingen.de/pub/Mirrors/download.mapsforge.org/maps/v5/), then configured the Obsidian Map View plugin to use 'http://localhost:8080/{z}/{x}/{y}.png' as it's map tile source. I'm currently working on a python app that will ingest OSMAnd files (markers, pictures, A/V notes, etc.) uploaded from my phone and produce .md notes for everything in my Obsidian vault. That allows all of my OSMAnd artifacts to be display on a map, and I can use Obsidian to add notes, link things together, etc. The one gap that currently exists is that Obsidian Map View doesn't support routes directly.

**Reply by dbogdanov** [reply]
@johnmcd88 Any further updates on your approach? I am considering something similar.

**Comment by Sonwon1** [comment]
I no longer recommend Furkot and have switched over to Kurviger for desktop route creation. It exports / imports well into Osmand. Kurviger works almost as well as Google Maps for finding stuff and when it doesn't I use Maps and then use the GPS coordinates to find the location in Kurviger. Kurviger's route creation is superior to Maps and can then be exported. There is a workaround for Osmand's lack of being able to import routes (track files work however route files just do not!) too. There are useful tools in Kurviger for nav routing too. Once Osmand's website catches up and becomes useful I will likely switch until then it is Kurviger on the desktop.

**Comment by thompehw** [comment]
Not helpful for Windows, but under Ubuntu Linux the Waydroid android system  is great. Every android app I've tried runs fine, including OsmAnd~ . I found it helpful to use another app to set a "fake" location (need to enable in android developer settings) since Waydroid doesn't otherwise supply OsmAnd with the equivalent of GPS.

**Comment by mariush444** [comment]
@vshcherb 
Maybe we can redefine problem. As I see many people can create convertes to/from gmaps, garmin and I see now with Obsydian. Transfering data is quite easy but the problem (for me) is with synchronisation (duplicates and so on going to and back).
Probably any unique id for wpt trk ... could be solution.
Just thinking, idea looking for solution.
